mlrun/datastore/base.py

```diff
@@ -157,6 +157,19 @@ def query(self, key, query="", **kwargs):
     def put(self, key, data, append=False):
         pass
 
+    def _prepare_put_data(self, data, append=False):
+        mode = "a" if append else "w"
+        if isinstance(data, bytearray):
+            data = bytes(data)
+        if isinstance(data, bytes):
```

Comment:
Better separate if from if-elif-else block to avoid it being misread as part of the same block.
```suggestion
            data = bytes(data)

        if isinstance(data, bytes):
```